test/unit-tests/components/views/rooms/UserIdentityWarning-test.tsx

```diff
@@ -0,0 +1,350 @@
+    });
+
+    // We don't display warnings in non-encrypted rooms, but if encryption is
+    // enabled, then we should display a warning if there are any users whose
+    // identity need accepting.
+    it("displays pending warnings when encryption is enabled", async () => {
+        room.getEncryptionTargetMembers = jest.fn(async () => {
+            return [mockRoomMember("@alice:example.org", "Alice")];
+        });
+        // Start the room off unencrypted.  We shouldn't display anything.
+        room.hasEncryptionStateEvent = jest.fn(() => false);
+        const crypto = client.getCrypto()!;
+        crypto["getUserVerificationStatus"] = jest.fn(async () => {
+            return Promise.resolve(new UserVerificationStatus(false, false, false, true));
+        });
+
+        render(<UserIdentityWarning room={room} key={ROOM_ID} />, {
+            wrapper: ({ ...rest }) => <MatrixClientContext.Provider value={client} {...rest} />,
+        });
+        await sleep(10); // give it some time to finish initialising
+        expect(() => getWarningByText("Alice's (@alice:example.org) identity appears to have changed.")).toThrow();
+
+        // Encryption gets enabled in the room.  We should now warn that Alice's
+        // identity changed.
+        client.emit(
+            RoomStateEvent.Events,
+            new MatrixEvent({
+                event_id: "$event_id",
+                type: EventType.RoomEncryption,
+                state_key: "",
+                content: {
+                    algorithm: "m.megolm.v1.aes-sha2",
+                },
+                room_id: ROOM_ID,
+                sender: "@alice:example.org",
+            }),
+            dummyRoomState(),
+            null,
+        );
+        await waitFor(() =>
+            expect(
+                getWarningByText("Alice's (@alice:example.org) identity appears to have changed."),
+            ).toBeInTheDocument(),
+        );
+    });
+
+    // When a user's identity needs approval, or has been approved, the display
+    // should update appropriately.
+    it("updates the display when identity changes", async () => {
+        room.getEncryptionTargetMembers = jest.fn(async () => {
+            return [mockRoomMember("@alice:example.org", "Alice")];
+        });
+        room.getMember = jest.fn(() => mockRoomMember("@alice:example.org", "Alice"));
+        const crypto = client.getCrypto()!;
+        crypto["getUserVerificationStatus"] = jest.fn(async () => {
+            return Promise.resolve(new UserVerificationStatus(false, false, false, false));
+        });
+        render(<UserIdentityWarning room={room} key={ROOM_ID} />, {
+            wrapper: ({ ...rest }) => <MatrixClientContext.Provider value={client} {...rest} />,
+        });
+        await sleep(10); // give it some time to finish initialising
+        expect(() => getWarningByText("Alice's (@alice:example.org) identity appears to have changed.")).toThrow();
```

Comment:
```suggestion
       await waitFor(() =>  expect(() => getWarningByText("Alice's (@alice:example.org) identity appears to have changed.")).toThrow());
```

Reply:
I don't think we need the `waitFor` here.  The component's initial state will cause the expectation to succeed.  It's only after the component has finished initialising that it could fail.  And if it fails, then waiting isn't going to ever make it correct.  So we aren't really waiting for the condition to succeed.  We're just checking whether it succeeds at that particular point in time.

We could add the `waitFor` to be consistent, but I think it's unnecessary, and would just add extra time to the test if the condition did fail.